src/components/FilesTable/FilesTable.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  // Button,
+  Table,
+} from 'react-bootstrap'
+
+// TODO(summer-cook):
+// add back handleDeleteFile to props once posting attachments/messages is working/during the work for this ticket:
+// https://github.com/scientist-softserv/webstore-component-library/issues/152*/
+const FilesTable = ({ addClass, files }) => {
+  if (files.length === 0) {
+    return (
+      <h6 className='mt-3'>You do not have this type of document yet.</h6>
+    )
+  }
+  return (
+    <Table striped bordered hover size='sm' className={`${addClass}`}>
+      <thead>
+        <tr>
+          <th>File Name</th>
+          <th>Uploaded By</th>
+          <th>Size</th>
+          <th>Created At</th>
+          {/* TODO(summer-cook):
+            - comment this back in, see comment above */}
+          {/* <th> </th> */}
+        </tr>
+      </thead>
+      <tbody>
+        {files.map((file) => {
+          const { uuid, fileName, uploadedBy, contentLength, createdAt } = file
+          return (
+            <tr key={uuid} className='small'>
+              <td>{fileName}</td>
+              <td>{uploadedBy}</td>
+              <td>{contentLength}</td>
+              <td>{createdAt}</td>
+              {/* TODO(summer-cook):
+                  - comment this back in (see above comment)
+                  - add an X icon here instead of remove text, make sure the handleDeleteFile is working */}
+              {/* <td>
+                    <Button
+                      primary
+                      onClick={handleDeleteFile}>
+                        Remove
+                    </Button>
+                  </td> */}
+            </tr>
+          )
+        })}
+      </tbody>
+    </Table>
+  )
+}
+
+FilesTable.propTypes = {
+  addClass: PropTypes.string,
+  files: PropTypes.arrayOf(
+    PropTypes.shape({
+      contentLength: PropTypes.string.isRequired,
+      contentType: PropTypes.string.isRequired,
+      createdAt: PropTypes.string.isRequired,
+      download: PropTypes.string.isRequired,
+      fileName: PropTypes.string.isRequired,
+      href: PropTypes.string.isRequired,
+      status: PropTypes.string,
+      uploadedBy: PropTypes.string.isRequired,
+      uuid: PropTypes.string.isRequired,
+    }),
+  ).isRequired,
+  // TODO(summer-cook): add this back in. see comment above.
+  // handleDeleteFile: PropTypes.func.isRequired,
+}
+
+FilesTable.defaultProps = {
+  addClass: '',
+  files: [{
+    status: null
+  }]
```

Comment:
files is being passed down from the filteredFiles in the viewFiles component. since filteredFiles is always an array, if it is an empty array because all the files of a certain type were filtered out, `files.length === 0` will be true. 
you can see in the story, there are no payment status files, so files is passed as an empty array, and the `files.length === 0` condition is satisfied.
<img width="1244" alt="image" src="https://user-images.githubusercontent.com/73361970/217391650-9c9846e4-a0a9-40a7-bc00-ddd4d737da26.png">

The status in the marketplace for some files IS null. So I'm not sure of a better way to communicate that.

Reply:
see if something like the below may work?

``` js
PropTypes.oneOfType([
  PropTypes.string.isRequired,
  PropTypes.oneOf([null]).isRequired,
]).isRequired
```

Reply:
I tried this and also 
`PropTypes.instanceOf(null)` which was in another answer, and both of those fail for me unfortunately.

But, i also tried the solution here: https://stackoverflow.com/questions/46839383/how-to-specify-proptypes-to-allow-nullable-required-prop-value

and it worked great. should I push that up? Then i can remove the defaultProps and make status required throughout.